ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -579,4 +586,5 @@ generatePrerequisites ghcFlavor = do
   -- of the way.
   removeFile "compiler/parser/Lexer.x"
```

Comment:
If we factor the generated files into a separate list as I suggested above we could also reuse it here instead of replicating the logic of which files are generated for which GHC version.

Reply:
Yes. Noted. As stated above, I'll raise an issue to clean this up.

Reply:
Fix coming.